### PR TITLE
Dev add px tokens

### PR DESCRIPTION
**Pull request by alexbpx**

### CODE REVIEW

File: src/main/java/com/perimeterx/internals/cookie/CookieData.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package com.perimeterx.internals.cookie;
+
+import lombok.Builder;
```
Comment (yaronpx) [inline src/main/java/com/perimeterx/internals/cookie/CookieData.java]: why do we need these annotations?

File: src/main/java/com/perimeterx/api/PerimeterX.java [hunk 2]
```diff
@@ -36,7 +39,7 @@
 import com.perimeterx.api.verificationhandler.DefaultVerificationHandler;
 import com.perimeterx.api.verificationhandler.VerificationHandler;
 import com.perimeterx.http.PXHttpClient;
-import com.perimeterx.internals.PXCaptchaValidator;
+import com.perimeterx.internals.PXCaptchaVerifier;
```
Comment (yaronpx) [inline src/main/java/com/perimeterx/api/PerimeterX.java]: why did you rename it?

File: src/main/java/com/perimeterx/api/PerimeterX.java [hunk 3]
```diff
@@ -204,6 +185,25 @@ public PXContext pxVerify(HttpServletRequest req, HttpServletResponseWrapper res
         return context;
     }
 
+    private void verifyCookie(PXContext context) throws Exception {
```
Comment (yaronpx) [inline src/main/java/com/perimeterx/api/PerimeterX.java]: 1. this is misleading, because this functions verifies multiple cookies (captcha, px) + does risk api if needed. 
2. it's bad practice to throw the general `Exception`. you should be more specific with the exceptions.

I really don't see the benefit of this refactoring

File: src/main/java/com/perimeterx/internals/CookieSelector.java [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+import java.util.Map;
+import java.util.Set;
+
+
+public class CookieSelector {
+
+    private static final PXLogger logger = PXLogger.getLogger(CookieSelector.class);
+
+    /**
+     * @param context - This context should contain the authorization header cookie and/or tokens header
```
Comment (yaronpx) [inline src/main/java/com/perimeterx/internals/CookieSelector.java]: missing param documentation

File: src/main/java/com/perimeterx/internals/CookieSelector.java [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+import com.perimeterx.models.risk.S2SCallReason;
+import com.perimeterx.utils.PXLogger;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+
+
+public class CookieSelector {
```
Comment (yaronpx) [inline src/main/java/com/perimeterx/internals/CookieSelector.java]: documentation needed. what is this class responsible for?

File: src/main/java/com/perimeterx/internals/CookieSelector.java [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+                    result = selectedCookie;
+                    break;
+                }
+            }
+        }
+        if (result == null && originalTokenCookie != null){
+            context.setOriginalToken(originalTokenCookie.getSelectedCookie());
+            Set<String> cookieKeys = originalTokenCookie.getCookieMap().keySet();
+            Iterator<String> versionsIter = cookieKeys.iterator();
+            if(versionsIter.hasNext()){
```
Comment (yaronpx) [inline src/main/java/com/perimeterx/internals/CookieSelector.java]: be consistent with code style. use spaces after keywords like `if`, `for`, etc.

**Comment by yaronpx** [comment]
there are a lot of code style issues all over this PR:
1. missing spaces after keywords and commas
2. too many empty lines (a good example is [this file](https://github.com/PerimeterX/perimeterx-java-sdk/pull/54/files#diff-9bd8a7e1740b2d2d7841cf0041574f3e))

File: src/main/java/com/perimeterx/models/PXContext.java [hunk 7]
```diff
@@ -266,174 +293,11 @@ private String extractCookieByKey(String cookie, String key) {
...
-            else if (cookieParts.length == 2) {
-                cookieVersion = AbstractPXCookie.getMobileCookieVersion(cookieParts[0]);
-                cookieMap.put(cookieVersion, cookieParts[1]);
-            }
-        }
-
-        return cookieMap;
-    }
-
-    public String getPxCookie() {
```
Comment (yaronpx) [inline src/main/java/com/perimeterx/models/PXContext.java]: a lot of deleted getters and setters here. never used them?

File: src/main/java/com/perimeterx/api/PerimeterX.java (hunk 3 above)
Comment (alexbpx) [reply src/main/java/com/perimeterx/api/PerimeterX.java]: 1. I'll rename it.
2. The general exception is thrown from shouldReverseRequest, which uses ReverseProxy.reversePxClient , that throws Exception as part of it's interface.

File: src/main/java/com/perimeterx/internals/CookieSelector.java (hunk 5 above)
Comment (alexbpx) [reply src/main/java/com/perimeterx/internals/CookieSelector.java]: It's responsible for selecting the right cookie before deserializing.
Meaning, if we have x-px-tokens and x-px-authorization headers, it tries to select pv3 from x-px-tokens. if it fails its fallback is pv1, and if that fails, it tries to use x-px-authorization.

I'll add the documentation

File: src/main/java/com/perimeterx/internals/CookieSelector.java (hunk 6 above)
Comment (alexbpx) [reply src/main/java/com/perimeterx/internals/CookieSelector.java]: !

File: src/main/java/com/perimeterx/models/PXContext.java (hunk 7 above)
Comment (alexbpx) [reply src/main/java/com/perimeterx/models/PXContext.java]: I've deleted a lot of fields that are no longer used after my refactoring

File: src/main/java/com/perimeterx/api/PerimeterX.java (hunk 3 above)
Comment (yaronpx) [reply src/main/java/com/perimeterx/api/PerimeterX.java]: well, if you're looking for more stuff to refactor...
